backend/mod.ts

```diff
@@ -67,25 +84,151 @@ router.post("/new", async (ctx) => {
       servers.push(serverId);
     };
 
-    for (let i = 1; i <= config.REPLICAS_RAID1; i++) {
+    for (let i = 1; i <= 3; i++) {
       await uploadToRandomServer();
     }
-  }
 
-  await addFileData({
-    id: objectId,
-    policy: parsedBody.data.policy,
-    filename: parsedBody.data.filename,
-    servers,
-    version: "v0",
-  });
+    console.log("Upload content RAID1:", parsedBody.data.filename);
+
+    await addFileData({
+      id: objectId,
+      policy: parsedBody.data.policy,
+      filename: parsedBody.data.filename,
+      servers,
+      version: "v0",
+      orders: [servers[0], servers[1], servers[2]],
+    });
+
+  } else if (parsedBody.data.policy === "RAID5"){
+    const b64 = parsedBody.data.content
+    const xor = B64XorCipher
+
+    console.log(b64.length)
+
+    const piece = b64.length/2
+
+    var A = b64.slice(0,piece)
+    var B = b64.slice(piece, b64.length)
+    var A_xor_B = xor.encode(A, B)
+
+    const uploadToRandomServer = async (i : number) => {
+      var serverId = getRandomInt(1, 6);
+      while (servers.indexOf(serverId) >=0){
+        serverId = getRandomInt(1, 6);
+      }
+      servers.push(serverId);
 
+      if (i == 1 ){
+        await upload(
+          objectId,
+          A,
+          `http://objectserver${serverId}:1040`,
+        );
+      } else if (i == 2){
+        await upload(
+          objectId,
+          B,
+          `http://objectserver${serverId}:1040`,
+        );
+      } else {
+        await upload(
+          objectId,
+          A_xor_B,
+          `http://objectserver${serverId}:1040`,
+        );
+      }
+    };
+
+    for (let i = 1; i <= 3; i++) {
+      uploadToRandomServer(i);
+    }
```

Comment:
Since we need to upload data to 3 independent servers, I suggest we can set "number of servers" of uploadToRandomServer instead of an index